packages/vats/src/core/chain-behaviors.js

```diff
@@ -382,6 +383,79 @@ export const makeChainStorage = async ({
   chainStorageP.resolve(rootNodeP);
 };
 
+/**
+ * @typedef HighPrioritySendersManager
+ * @property {(namespace: string, address: string) => Promise<void>} add
+ * @property {(namespace: string, address: string) => void} remove
+ */
+export const PRIORITY_SENDERS_NAMESPACE_RE = /^[a-zA-Z0-9_-]{1,50}$/;
+
+/**
+ * @param {BootstrapSpace & {
+ *   consume: { loadCriticalVat: ERef<VatLoader<ChainStorageVat>> }
+ * }} powers
+ */
+export const makeHighPrioritySendersManager = async ({
+  consume: { loadCriticalVat, bridgeManager: bridgeManagerP },
+  produce: { highPrioritySendersManager: managerP },
+}) => {
+  const bridgeManager = await bridgeManagerP;
+  if (!bridgeManager) {
+    console.warn('Cannot support chainStorage without an actual chain.');
+    managerP.resolve(null);
+    return;
+  }
+
+  const storageBridgeManager = E(bridgeManager).register(BRIDGE_ID.STORAGE);
+
+  /** @type {Map<string, StorageNode>} */
+  const addressNodes = new Map();
+
+  const vat = E(loadCriticalVat)('bridge');
+  const sendersNode = E(vat).makeBridgedChainStorageRoot(
+    storageBridgeManager,
+    STORAGE_PATH.HIGH_PRIORITY_SENDERS,
+    { sequence: true },
+  );
+
+  const manager = Far('highPrioritySenders manager', {
```

Comment:
We're trying to stop exporting objects from bootstrap, right?

 - #4548

Reply:
Correct. As is this doesn't have precious state because it's trivially recomposed from vstorage.

Given the requirement Mathieu describes it may need to have state that isn't expressed on chain. In that case it would need durable storage.

Reply:
There's also a goal of not sending messages to the bootstrap vat at all during normal operation... that is: after bootstrap and other than core eval. But I suppose that's not critical.

Reply:
I think that sending a message to the boostrap vat when there is a fundamental change made to the underlying layer of the chain is an acceptable exception.

Reply:
So thinking more about this non durable `Far` thing, if we ever need to upgrade the bootstrap vat, we could reconstitute the state from vstorage if needed, but the manager object reference shared with other vats would be broken anyway.

What would be the plan to re-distribute this manager?

Reply:
Documented in https://github.com/Agoric/agoric-sdk/pull/7500/files#diff-b880209dcc8e32e40e53c7e255d9c942b08ad7a36638680de7bd7b0f73b42a6dR414-R420